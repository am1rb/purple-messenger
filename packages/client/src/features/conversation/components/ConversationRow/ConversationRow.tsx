import React, { memo } from "react";
import clsx from "clsx";
import { Conversation, MessageOwner } from "@purple-messenger/core";
import { Card, CardContent, Box, Avatar, Typography } from "@material-ui/core";
import MessageStatus from "features/message/components/MessageStatus";
import MessageSentAt from "features/message/components/MessageSentAt";
import CardActionArea from "components/CardActionArea";
import iff from "core/helper/iff";
import useStyles from "./ConversationRow.styles";

export interface ConversationRowProps extends Conversation {
  selected: boolean;
}

function ConversationRow({
  message,
  friend,
  selected,
  isTyping,
}: ConversationRowProps) {
  const classes = useStyles();

  return (
    <Card className={clsx(classes.root, iff(selected, classes.selected))}>
      <CardActionArea to={"/conversation/@" + friend.username} disableRipple>
        <CardContent>
          <Box display="flex" alignItems="center" width="100%">
            <Avatar src={friend.image} />
            <Box ml={1} overflow="hidden" width="100%">
              <Box>
                <Typography noWrap>
                  {friend.firstName} {friend.lastName}
                </Typography>
                <Typography
                  variant="subtitle2"
                  noWrap
                  color="textSecondary"
                  gutterBottom
                >
                  {message?.owner === MessageOwner.Me && "You: "}
                  {isTyping ? "Typing..." : message?.body || "No message"}
                </Typography>
              </Box>
              {message && (
                <Box display="flex" justifyContent="space-between" width="100%">
                  <MessageSentAt sentAt={message.sentAt} />
                  <MessageStatus status={message.status} />
                </Box>
              )}
            </Box>
          </Box>
        </CardContent>
      </CardActionArea>
    </Card>
  );
}

export default memo(ConversationRow);
