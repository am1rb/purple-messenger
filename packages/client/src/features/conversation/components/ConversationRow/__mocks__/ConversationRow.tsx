import React from "react";

export interface ConversationRowProps {
  selected: boolean;
}

function ConversationRow({ selected }: ConversationRowProps) {
  return (
    <div data-testid="mock-conversation-row">
      {selected && <span data-testid="mock-conversation-row-selected" />}
    </div>
  );
}

export default ConversationRow;
