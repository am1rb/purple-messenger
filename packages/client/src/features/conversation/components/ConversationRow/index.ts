export * from "./ConversationRow";
export { default } from "./ConversationRow";
