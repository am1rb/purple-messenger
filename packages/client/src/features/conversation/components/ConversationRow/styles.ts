import { makeStyles, Theme } from '@material-ui/core';

export default makeStyles((theme: Theme) => ({
  root: { marginBottom: theme.spacing(1) },
  selected: { backgroundColor: theme.palette.secondary.light },
}));