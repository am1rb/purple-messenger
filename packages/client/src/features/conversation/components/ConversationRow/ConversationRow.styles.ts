import { Theme } from "@material-ui/core";
import { makeStyles } from "@material-ui/styles";

export default makeStyles((theme: Theme) => ({
  root: { marginBottom: theme.spacing(1) },
  selected: { backgroundColor: theme.palette.secondary.light },
}));
