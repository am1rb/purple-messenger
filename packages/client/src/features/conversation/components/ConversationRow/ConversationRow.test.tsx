import React from "react";
import { render } from "@testing-library/react";
import {
  AccountStatus,
  MessageOwner,
  MessageStatus,
} from "@purple-messenger/core";
import ConversationRow, { ConversationRowProps } from ".";

jest.mock("features/message/components/MessageStatus");
jest.mock("features/message/components/MessageSentAt");
jest.mock("components/CardActionArea");
jest.mock("features/message/components/MessageIsTyping");
jest.mock("../ConversationNoMessage");

const sharedProps: ConversationRowProps = {
  selected: true,
  id: 1,
  isTyping: true,
  friend: {
    id: 1,
    username: "user1",
    bio: "lumuiwuamielgiwonorrapkufpicagfevusevagivcistopkag",
    email: "jubsinbam@lej.uk",
    firstName: "Lydia",
    lastName: "Guzman",
    image: "",
    status: AccountStatus.Online,
  },
};

describe("The <ConversationRow /> tests", () => {
  it("Should match the snapshot", () => {
    const { container } = render(<ConversationRow {...sharedProps} />);
    expect(container).toMatchSnapshot();
  });

  it("Should render the sent at and status if a message object is passed", () => {
    const { getByTestId } = render(
      <ConversationRow
        {...sharedProps}
        message={{
          id: 1,
          body: "This is the message body",
          owner: MessageOwner.Me,
          sentAt: new Date(),
          status: MessageStatus.Seen,
          unreadCount: 1,
        }}
      />
    );
    expect(getByTestId("message-status")).toBeInTheDocument();
    expect(getByTestId("message-sent-at")).toBeInTheDocument();
  });
});
