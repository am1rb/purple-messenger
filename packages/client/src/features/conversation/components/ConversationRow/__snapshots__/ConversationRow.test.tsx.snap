// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The <ConversationRow /> tests Should match the snapshot 1`] = `
<div>
  <div
    data-testid="mock-mui-card"
  >
    <div
      data-testid="mock-card-action-area"
    >
      <div
        data-testid="mock-mui-card-content"
      >
        <div
          data-testid="mock-mui-box"
        >
          <img
            data-testid="mock-mui-avatar"
            src=""
          />
          <div
            data-testid="mock-mui-box"
          >
            <div
              data-testid="mock-mui-box"
            >
              <div
                data-testid="mock-mui-typography"
              >
                Lydia
                 
                Guzman
              </div>
              <div
                data-testid="mock-mui-typography"
              >
                <span
                  data-testid="mock-message-is-typing"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
